base/sharedarray.jl

```diff
@@ -249,7 +249,7 @@ function rand!{T}(S::SharedArray{T})
 end
 
 function randn!(S::SharedArray)
-    f = S->map!(x->randn, S.loc_subarr_1d)
+    f = S->map!(x->randn(), S.loc_subarr_1d)
```

Comment:
It seems to be only defined for `AbstractArray{Float64}` though, while it would be nice to keep the `SharedArray` code generic (for example, in the previous version it was generically and uniformly failing... ehm).
See also #9836.

On the other hand, `rand!` should be fine AFAICT.